### ISSUE TITLE
Showing Discs and Album Size

### ISSUE THREAD

**Issue by chadfranklin47** [digimezzo/dopamine-windows#1064]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report  
[x] Feature request
</code></pre>

Hello again, I have noticed that when I have multiple discs in an album and when I look at the tracks in Dopamine, Dopamine indicates which tracks fall under which discs WHEN THE FILES ARE M4A. If the files are FLAC, then Dopamine just lists the files straight without indicating what disc they are in and it gets confusing. Is this by design? Can this be changed or fixed? Also, is it by design that a 2.83 GB album shows the size "2 GB" in dopamine? I think it would be better to round to the nearest whole number, but it's not that big of a deal.

Screenshots:
![image](https://user-images.githubusercontent.com/37273511/70847513-3f286a00-1e33-11ea-8f3c-c3cf9159ddf5.png)

vs

![image](https://user-images.githubusercontent.com/37273511/70847515-4b142c00-1e33-11ea-818b-64fceb082ace.png)

And in this case, the album songs add up to 2.81GB as shown here:
![image](https://user-images.githubusercontent.com/37273511/70847530-7dbe2480-1e33-11ea-8d4f-48f34a17d9be.png)

**Comment by digimezzo** [comment]
@chadfranklin47 Could you attach some screenshots to indicate where you are seeing and not seeing the disc info and wrong size calculation? Thanks!

**Comment by chadfranklin47** [comment]
Absolutely. These are in the album section after clicking on an album cover.

![image](https://user-images.githubusercontent.com/37273511/70847454-8a8e4880-1e32-11ea-8db0-3085eb19f856.png)

vs

![image](https://user-images.githubusercontent.com/37273511/70847461-9c6feb80-1e32-11ea-8700-7f71e1de2df6.png)

**Comment by chadfranklin47** [comment]
@digimezzo

**Comment by digimezzo** [comment]
@chadfranklin47 Are the "Disc number" tags filled in for all these files?

**Comment by chadfranklin47** [comment]
@digimezzo Yes, the disc numbers are filled in, at least according to Mp3tag. Also, this is the case for multiple, if not all, FLAC albums I have, not just one.

**Comment by digimezzo** [comment]
@chadfranklin47 Ok. Thanks for the info. I'll look at it.

**Comment by chadfranklin47** [comment]
@digimezzo Thank you.

**Comment by digimezzo** [comment]
@chadfranklin47 I fixed the album size bug. I'll be in the next release (which will be Beta 1). i'm now looking into the disc issue.

**Comment by digimezzo** [comment]
@chadfranklin47 The disc thing is not an issue. Please check the "Disc count" tag on your mp3 files. The discs are only split up when the disc count tag is set. Without that info, Dopamine cannot know there is more than 1 disc.

I'm guessing that your m4a files have the disc count set properly. Could you confirm this?

**Comment by chadfranklin47** [comment]
@digimezzo Just to reiterate, I have only ever gotten the bug with FLAC files. The files were tagged correctly, but it seems the bug has something to do with FLAC itself, maybe an older version of FLAC is buggy as converting all of my files from FLAC to FLAC using the newest version of foobar2000 fixes the problem. Here is a link to a short video displaying the problem files and the fixed ones. The only difference between them is the album name and me doing the conversion with foobar:

https://youtu.be/PHu94bR9uFs

**Comment by digimezzo** [comment]
@chadfranklin47 Thanks a lot for the extra info and video. That really helpful. I'm just wondering, can Dopamine see the disc number info? To check this, could you right click on one of the oolder flac files that still has that issue, and choose "Edit" (that should open the Dopamine tag editor). Could yo utell me if the disc number tag is filled in there? Thanks

**Comment by chadfranklin47** [comment]
@digimezzo Ok, I see it now. Dopamine can see the Disc Number for the old files but not the Disc Count. While with the new files, it sees both. But as far as I know, Disc Count isn't something that I set manually, is it? Mp3tag only has a tag for DiscNumber, not DiscCount. Same with Track count. And I didn't change anything for the new files, all I did was convert from FLAC to FLAC with the newest foobar2000. Here is the screenshot for the old files:

![image](https://user-images.githubusercontent.com/37273511/70869510-b6f0b480-1f59-11ea-9891-be52111f914b.png)

and the new files:

![image](https://user-images.githubusercontent.com/37273511/70869518-cb34b180-1f59-11ea-9635-3d10b00df331.png)

**Comment by chadfranklin47** [comment]
@digimezzo Is Disc Count and Track Count just a Dopamine thing or a thing embedded in the files? Is there a way for the code to not rely on those attributes? It seems doable with just the track number and the disc number.

**Comment by chadfranklin47** [comment]
@digimezzo Small update, I get the same error on one of my Ogg formatted albums.

**Comment by digimezzo** [comment]
@chadfranklin47 The choice of using the disc count instead of track number and disc number is related to performance. When using disc number, 2 passes over the whole list of selected tracks would be required to know if an album consists of more than 1 disc (Dopamine would have to check the disc numbers for all tracks of an album, and see if there are different disc numbers). Disc count immediately shows if there is more than 1 disc, on the 1st pass. That is much faster. So that logic will probably never change.

Disc count and disc number are indeed tags that are embedded in the files. I'm not sure what is wrong with the Flac and Ogg files though. And equally unsure why re-encoding flac files resolves the issue.
Would you mind sharing a flac and ogg file which still have this issue, so I could have a closer look?

**Comment by chadfranklin47** [comment]
Ok, I see @digimezzo . Absolutely. Here is a download link containing a zip file of one track from the FLAC album, one from the Ogg album, and the same FLAC track after re-encoding which fixes the problem:

https://mega.nz/#!PmwUXCCb!wKaE8o3QGxrwzftmTskpqMZ1ygV0ZmZiqpOQFCs8u4w

**Comment by digimezzo** [comment]
@chadfranklin47 Excellent, I've downloaded them. Thanks!
